### ISSUE TITLE
Train detection model with TWD database

### ISSUE THREAD

**Issue by diegoperezvieites** [Gradiant/ai-project-doc_layout_detection#1]
## Description

Data was explored throw other ai-project, for this reason we start directly with the train stage.

**Comment by diegoperezvieites** [comment]
I need some help: @mmeendez8 , @mbaptistagrd 

Several models were trained, but results are bad. The goal is to detect text blocks or paragrahps in text pages.

The anchors were calculated for train data, but the results are still bad : http://10.10.30.58:8999/#/experiments/63/runs/3fd5115d90914a218315ba648603743e

![overlap](https://user-images.githubusercontent.com/66787477/133057819-2c27d9fc-fabd-40e1-8562-afc5e3600b0d.png)

![clusters](https://user-images.githubusercontent.com/66787477/133059642-ba84fd62-f21c-4cde-8d83-04e06085f64e.png)

I saw data and annotations seems to be correct:

**(TRAIN ANNOT EXAMPLE)**
![Captura de pantalla de 2021-09-13 11-26-13](https://user-images.githubusercontent.com/66787477/133059483-a5057b0e-6fb8-4698-9a98-3f764b558bfc.png)


**(EVAL ANNOT EXAMPLE)**
![Captura de pantalla de 2021-09-13 11-12-17](https://user-images.githubusercontent.com/66787477/133057479-5f262258-5776-4558-b6be-ab998e828f10.png)


**(PREDICTION EXAMPLE)**
![0004 ag_mix_Medium-0 (1)](https://user-images.githubusercontent.com/66787477/133057637-ae46a4ff-5e2b-45c9-90f3-060212db7174.jpg)

**Comment by mmeendez8** [comment]
Check your original images size and then find a better value for your image_size argument:
Image size: https://github.com/Gradiant/ai-project-doc_layout_detection/blob/issue-1-Train_detection_model_with_TWD_database/configs/datasets/TWD.py#L9

I do not think a squared shape is what you are looking for

**Comment by diegoperezvieites** [comment]
No, It isnt! It is good to know the relevance of the input shape! I will try a better aspect ratio of the input image.

Thanks!

**Comment by diegoperezvieites** [comment]
> Check your original images size and then find a better value for your image_size argument:
> Image size: https://github.com/Gradiant/ai-project-doc_layout_detection/blob/issue-1-Train_detection_model_with_TWD_database/configs/datasets/TWD.py#L9
> 
> I do not think a squared shape is what you are looking for

The results are still bad. I tried using a segmentation project and the results are much better. I think the detection network should work too! I do not understand why this network does not work.

**OUTPUT DETECTION NET:**
![0004 ag_mix_Medium-0 (3)](https://user-images.githubusercontent.com/66787477/133215872-849f70ec-1c36-41b1-8402-9b925b0c514b.jpg)


**OUTPUT SEGMENTATION NET:**
![0004 ag_mix_Medium-0 (2)](https://user-images.githubusercontent.com/66787477/133215853-d5567f79-6545-458e-94bf-873c5e1a3742.jpg)

**Comment by mmeendez8** [comment]
Try to see how your data looks just before being feed to the network. You can use: 

https://github.com/Gradiant/ai-project-template/blob/master/ai-project-%7B%7Bcookiecutter.project_slug%7D%7D/stages/experiment/registry_extras/data_visualizer.py#L9

Can't remember if it should be located before or after DefaultFormatBundle:
https://github.com/Gradiant/ai-project-template/blob/master/ai-project-%7B%7Bcookiecutter.project_slug%7D%7D/configs/datasets/detection.py#L21
